hs_modelinstance/models.py

```diff
@@ -37,23 +40,40 @@ def remove(cls, element_id):
 
 class ExecutedBy(AbstractMetaDataElement):
     term = 'ExecutedBY'
-    name = models.CharField(max_length=500)
-    url = models.URLField()
+    name = models.CharField(max_length=500, choices=(('-','    '),))
+    model_program_fk = models.ForeignKey('hs_model_program.ModelProgramResource', null=True, blank=True)
+
 
     def __unicode__(self):
         self.name
 
     @classmethod
     def create(cls, **kwargs):
-        return ExecutedBy.objects.create(**kwargs)
+        shortid = kwargs['name']
+
+        obj = get_object_or_404(ModelProgramResource,short_id=shortid)
+        kwargs['model_program_fk'] = obj
+        metadata_obj = kwargs['content_object']
+        title = obj.title
+        mp_fk = ExecutedBy.objects.create(model_program_fk=obj,
+                                          name=title,
+                                          content_object=metadata_obj)
+
+        return mp_fk
+
```

Comment:
If you want to customize creation, you should customize `ExecutedBy.objects.create()` itself via a custom `ModelManager`